### ISSUE TITLE
work in node 4

### ISSUE THREAD

**Issue by ryanxcharles** [bitpay/bitcore#1361]
Awesome work with the new launch of bitcore 1.0. Unfortunately it doesn't work out of the box in node 4. bitcored gives this error:

<pre>
/home/ryan/dev/bitcore/node_modules/bitcore-node/index.js:8
    throw new Error('Node.js version ' + version + ' is expected to be ' + expected);
    ^

Error: Node.js version 4.2.1 is expected to be ^0.12
</pre>

I tried commenting out the version check, and I get a segmentation fault. bitcore-node also doesn't compile in node 4 and gives a huge number of errors. These problems are probably related to changes in the node C++ API between 0.12 and 4.0.

My platform is Linux Mint.

**Comment by braydonf** [comment]
Yep, the C++ bindings located at https://github.com/braydonf/bitcore-node/blob/node-v4/src/libbitcoind.cc need to be updated fully with Nan 2.0 so that it's possible to compile for multiple versions of Node.js. I've started [work on this](https://github.com/braydonf/bitcore-node/tree/node-v4) and have `getBlock` working, however there are some additional bindings that need to be updated.

Related issue: https://github.com/bitpay/bitcore-node/issues/259

**Comment by braydonf** [comment]
@ryanxcharles This PR ([and branch](https://github.com/braydonf/bitcore-node/tree/node-v4)) has been updated: https://github.com/bitpay/bitcore-node/pull/348 and should now compile with Node.js v4 _(tested with Ubuntu 14.04)_.

**Comment by ryanxcharles** [comment]
Nice!

**Comment by ryanxcharles** [comment]
I do in fact have bitcore running in node 4 right now, so thanks! However, it still doesn't actually work out of the box. bitcore master pulls an old version of bitcore-node that isn't fixed. I then tried to download bitcore-node (the master of which does, in fact, compile correctly!) and then use npm link to link them. This actually doesn't work either, because bitcore complains that there are now two instances of bitcore-lib. I finally got it to work by compiling bitcore-node, and copying the entire directory into bitcore/node_modules, and then deleting the instance of bitcore-lib inside bitcore-node/node_modules. So, basically, if you guys updated version numbers for bitcore-node and bitcore, I think that would fix things :)
